tools/scylla-nodetool.cc

```diff
@@ -1751,22 +1752,53 @@ void restore_operation(scylla_rest_client& client, const bpo::variables_map& vm)
         }
         params[required_param] = vm[required_param].as<sstring>();
     }
-    if (!vm.contains("sstables")) {
-      throw std::invalid_argument("missing required possitional argument: sstables");
+    bool sstables_as_params = vm.contains("sstables");
+    bool sstables_as_file_list = vm.contains("sstables-file-list");
+    if (not sstables_as_params and not sstables_as_file_list) {
+      throw std::invalid_argument("missing both argument: sstables and --sstables-file-list (at least one is required)");
     }
     if (vm.contains("scope")) {
         params["scope"] = vm["scope"].as<sstring>();
     }
-    sstring sstables_body = std::invoke([&vm] {
+
+    std::vector<sstring> sstables_list;
+    
+    // add the list given by the file param
+    if (sstables_as_file_list) {
+        sstring sstables_list_file = vm["sstables-file-list"].as<sstring>();
+        std::ifstream ifs(sstables_list_file);
```

Comment:
Please use seastar io, you can conveniently read the entire file content with `util::read_entire_stream_contiguous()` into a single string, then split with `std::views::split()`.

Reply:
> Please use seastar io, you can conveniently read the entire file content with `util::read_entire_stream_contiguous()` into a single string, then split with `std::views::split()`.

Are we ok with allocating several megabytes at once to read this file? and several could be 20?

Reply:
with non-contiguous version you will have to sew lines manually which would add unnecessary complication to such a simple task, I guess